stm32/unittest/scripts/tst_api_demod_check

```diff
@@ -37,24 +50,81 @@ case "${TEST_OPT}" in
+	p1=$(grep '^BER\.*: 0.000' stderr.log | wc -l)
+	p2=$(grep '^Coded BER: 0.000' stderr.log | wc -l)
 	if [[ $p1 -eq 1 && $p2 -eq 1 ]]; then echo "OK"; 
 	else echo "BAD"; 
      	    let Fails=($Fails + 1)
      	fi
 	;;
 
-    700D_plain_codec)
-        echo -e "\nCompare output binary data"
-        if compare_ints -s -b2 ref_mod.raw stm_out.raw; then 
-            echo "Passed"
-        else
-            echo "Failed"
-            let Fails=($Fails + 1)
+    700D_AWGN_test)
+	echo "Check reference decode"
+	uber_ref=$(cat ref_gen.log | sed -n "s/^BER.* \([0-9..]*\) Tbits.*/\1/p")
+	cber_ref=$(cat ref_gen.log | sed -n "s/^Coded BER.* \([0-9..]*\) Tbits.*/\1/p")
+	printf "REF   uncoded BER: %f coded BER: %f\n" $uber_ref $cber_ref
+
+	# As per notes in tst_api_demod_setup, coded BER is unreliable
+	# for such a short test, so we'll just sanity check the
+	# reference uncoded BER here. Bash can't compare floats
+	# .... so use return code of some python script
+	python -c "import sys; sys.exit(1) if $uber_ref<0.1 else sys.exit(0)"
+	if [[ $? -eq 1 ]]; then echo "OK"; 
+	else echo "BAD"; 
+     	    let Fails=($Fails + 1)
+     	fi
+
+	echo "Check target decode"
+	uber_target=$(cat ref_gen.log | sed -n "s/^BER.* \([0-9..]*\) Tbits.*/\1/p")
+	cber_target=$(cat ref_gen.log | sed -n "s/^Coded BER.* \([0-9..]*\) Tbits.*/\1/p")
+	printf "TARGET uncoded BER: %f coded BER: %f\n" $uber_target $cber_target
+	python -c "import sys; sys.exit(1) if $uber_target<0.1 and abs($cber_ref-$cber_target)<0.01 else sys.exit(0)"
+	if [[ $? -eq 1 ]]; then echo "OK"; 
+	else echo "BAD"; 
+     	    let Fails=($Fails + 1)
      	fi
 	;;
 
+    700D_AWGN_codec)
+	# 1/ The two output files sound OK, and when plotted look very
+	#    similar, but they don't match on a sample-sample basis.
+	
+	# 2/ Suspect some small state difference, or perhaps random
+	#    number generator diverging, sampling codec2_rand() at the
+	#    end of the x86 and stm32 test programs might show up any
+	#    differences.
+
+	# 3/ At this stage - we can't make sample by sample automatic
+	#    tests work.  However there is value in running the test
+	#    to ensure no asserts are hit and the code doesn't crash
+	#    (e.g. due to an out of memory issues). A simple energy
+	#    comparison is used on the output speech files, which
+	#    will trap any large errors.
+
+	# 4/ We can also manually evaulate the ouput decoded speech by
+	#    listening to the output speech files.
+
+	compare_energy;
+
+	# make sure execution time stays within bounds
+	execution_time=mktmp
+	cat stdout.log | sed  -n "s/.*freedv_rx \([0-9..]*\) msecs/\1/p" > $execution_time
+	python -c "import sys; import numpy as np;  x=np.loadtxt(\"$execution_time\"); print(\"execution time max:: %5.2f mean: %5.2f ms\" % (np.max(x), np.mean(x))); sys.exit(1) if np.max(x) < 80.0 else sys.exit(0)"
+	if [[ $? -eq 1 ]]; then echo "execution time OK"; 
+        else echo "BAD"; 
+     	   let Fails=($Fails + 1)
+        fi
```

Comment:
@donmr I added code to check execution time.  Had to place a limit on LPDC decoder iterations to get it into real time.